src/targets/gpu/include/migraphx/gpu/compile_pointwise.hpp

```diff
@@ -11,11 +11,14 @@ struct module;
 namespace gpu {
 
 struct context;
-operation compile_pointwise(context& ctx,
-                            const std::vector<shape>& inputs,
-                            const std::string& lambda,
-                            const std::string& preamble = "");
-
+operation
+compile_pointwise(context& ctx,
+                  const std::vector<shape>& inputs,
+                  const std::string& lambda,
+                  const std::string& preamble = "",
+                  int global_workitems = 1024, // Global (total) work items. todo: testing indicates
+                                               // values of global close to tensor size are best
+                  int local_workitems = 256);  // Local work items per CU
```

Comment:
Dont provide these defaults here. Instead add a function overload with these parameters missing and call `compute_global` with local set to 1024. Otherwise, there will be a performance regression.

Reply:
This change has been made, commit d66690b22f1.  Please review as it's rather mistake-prone.  Also, I had to rearrange argument list with preamble last.